privacyidea/lib/token.py

```diff
@@ -531,6 +532,18 @@ def get_tokens_paginate(tokentype=None, realm=None, assigned=None, user=None,
                 log.debug(traceback.format_exc())
                 token_dict["username"] = "**resolver error**"
 
+            if excluded_tokeninfo_fields:
+                # we cant simply delete the key-value pair because than we would change the size of the dict during
+                # the iteration
+                tmp = {}
+                # Loop over all the key-value pairs and check if they are not excluded. if there are not in the
+                # exclusion-list we add them to the temp-dict.
+                for key in token_dict['info'].keys():
+                    if key not in excluded_tokeninfo_fields:
+                        tmp[key] = token_dict['info'][key]
+
+                # now we just replace the old info-dict with the newly created one.
+                token_dict['info'] = tmp
```

Comment:
The error occurs if you change the size of the dict keys during iteration. You can circumvent this by this code snipped and avoid your longer workaround:

```python
                for key in list(token_dict['info']):
                    if key in excluded_tokeninfo_fields:
                        token_dict['info'].pop(key)
```
You iterate not over the `.keys` themselves, but in the copied `list(token_dict['info'])`, which works.